server/weaverbird/backends/sql_translator/steps/top.py

```diff
@@ -40,20 +40,22 @@ def translate_top(
 
     # We build the group by query part
     group_by_query: str = ''
-    for index, gb in enumerate(step.groups + [step.rank_on]):
-        group_by_query += ('GROUP BY ' if index == 0 else ', ') + gb
+    if len(step.groups) > 0:
+        for index, gb in enumerate(step.groups + [step.rank_on]):
+            group_by_query += ('GROUP BY ' if index == 0 else ', ') + gb
 
-    # We remove superflues columns
-    # to handle upper and lower cases
-    # we refresh query metadatas with columns to not remove
-    cols = [g.upper() for g in step.groups] + [step.rank_on.upper()]
-    cols_to_remove = [
-        col
-        for col in query.metadata_manager.retrieve_query_metadata_columns_as_list()
-        if col not in cols
-    ]
-    query.metadata_manager.remove_query_metadata_columns(cols_to_remove)
+    # We remove other columns not in the group-by
```

Comment:
In the pandas implementation, columns arent removed.
So you keep all columns of the top rows.

Reply:
Good point, because if we use `Group By` in SQL we'll lose columns anway. So to have the same behavior as in Pandas, we'll need to perform an auto-join a remove non aggregated columns.